Sources/Sentry/SentryHttpStatusCodeRange.m

```diff
@@ -0,0 +1,32 @@
+#import <SentryHttpStatusCodeRange.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation SentryHttpStatusCodeRange
```

Comment:
@philipphofmann `NSRange` is a struct and can't be used with an Array, also, NSRange isn't `from` -> `to` but rather `init` and `how many items`, so @brustolin said that this would be better, and more compatible with the spec.

Reply:
Shitty ObjC 🙈. You can put an NSRange into an `NSArray` by using `NSValue`, but for the options, that would look weird, as you could also put `NSInteger`, `Char`, or other stuff into it.  So we could do `@property (nonatomic, assign) NSArray<NSValue *> *failedRequestStatusCodes`, but then in Swift you could do `options.failedRequestStatusCodes = [NSRange(400..<410), 12, "k"]`, which is ugly.

I agree that creating a custom class for this makes sense. Please add a short decision entry here https://github.com/getsentry/sentry-cocoa/tree/master/develop-docs#decision-log on why we didn't use NSRange, and link to that decision from here.

Reply:
https://github.com/getsentry/sentry-cocoa/pull/2308/commits/88a035175d1e1df68a2446f531a625a856882e43